Sources/Core/GTMSessionFetcher.m

```diff
@@ -872,6 +872,26 @@ - (void)beginFetchMayDelay:(BOOL)mayDelay
     }
   }
 
+  BOOL stopped;
+  @synchronized(self) {
+    GTMSessionMonitorSynchronized(self);
+    stopped = _userStoppedFetching;
+  }
+  if (stopped) {
+    GTMSESSION_ASSERT_DEBUG(_delayState == kDelayStateNotDelayed,
+                            @"Unexpected internal state: %lu", (unsigned long)_delayState);
+    // We end up here if someone calls `stopFetching` from another thread/queue while
+    // the fetch was being started up, so while `stopFetching` did the needed shutdown
+    // we have to ensure the requested callback was triggered.
+    if (self.stopFetchingTriggersCompletionHandler) {
+      NSError *error = [NSError errorWithDomain:kGTMSessionFetcherErrorDomain
+                                           code:GTMSessionFetcherErrorUserCancelled
+                                       userInfo:nil];
```

Comment:
Do we ever bother with supplying any maybe useful userInfo like the URL being fetched (or cancelled in this case).

Reply:
Not currently, but I have a follow up item in my notes to maybe revisit this in the future.

(In most contexts the `fetcher` is also right there at the `begin...` call, so they could use it in their handler also)